src/pharmpy/statements.py

```diff
@@ -593,6 +590,38 @@ def set_lag_time(self, compartment, lag_time):
         return new_comp
 
 
+def _is_positive(expr: sympy.Expr) -> bool:
+    return sympy.ask(
+        sympy.Q.positive(expr), _assume_all(sympy.Q.positive, _free_images_and_symbols(expr))
+    )
+
+
+def _free_images_and_symbols(expr: sympy.Expr) -> Set[sympy.Expr]:
+    return expr.free_symbols | _free_images(expr)
+
+
+def _free_images(expr: sympy.Expr) -> Set[sympy.Expr]:
```

Comment:
This function only does `set`. Feels to little to warrant a separate function.

Reply:
What I don't like is to have this many small private function in the top level of statements. It is a bit noisy. Perhaps move them into the function that uses them or merge them into one single function.

Reply:
It's nice to have access to both an iterator and a set implementation. It also allows to use Python `yield` syntax without any explicit temporary collection object construction. `_free_images_iter` and `_free_images` end up being simpler to understand without any comment.

Reply:
To me it adds unnecessary noise since you only needed one of them now. Especially since this is in one of the core modules. If it is only the conversion function that needs these they could be moved inside it.